src/app/features/public-form/public-form.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component } from "@angular/core";
+import { AppSettings } from "../../core/app-config/app-settings";
+import { PouchDatabase } from "../../core/database/pouch-database";
+import PouchDB from "pouchdb-browser";
+import { ActivatedRoute } from "@angular/router";
+import { EntityRegistry } from "../../core/entity/database-entity.decorator";
+import { EntityMapperService } from "../../core/entity/entity-mapper.service";
+import { PublicFormConfig } from "./public-form-config";
+import { Entity, EntityConstructor } from "../../core/entity/model/entity";
+import { toFormFieldConfig } from "../../core/entity-components/entity-subrecord/entity-subrecord/entity-subrecord-config";
+import { FormFieldConfig } from "../../core/entity-components/entity-form/entity-form/FormConfig";
+import {
+  EntityForm,
+  EntityFormService,
+} from "../../core/entity-components/entity-form/entity-form.service";
+import { EntityFormComponent } from "../../core/entity-components/entity-form/entity-form/entity-form.component";
+import { MatButtonModule } from "@angular/material/button";
+import { ConfigService } from "../../core/config/config.service";
+import { EntitySchemaService } from "../../core/entity/schema/entity-schema.service";
+import { MatSnackBar } from "@angular/material/snack-bar";
+import { MatCardModule } from "@angular/material/card";
+
+@Component({
+  selector: "app-public-form",
+  templateUrl: "./public-form.component.html",
+  styleUrls: ["./public-form.component.scss"],
+  imports: [EntityFormComponent, MatButtonModule, MatCardModule],
+  standalone: true,
+})
+export class PublicFormComponent<E extends Entity> {
+  private entityType: EntityConstructor<E>;
+  private prefilled: Partial<E> = {};
+  formConfig: PublicFormConfig;
+  entity: E;
+  columns: FormFieldConfig[][];
+  form: EntityForm<E>;
+
+  constructor(
+    private database: PouchDatabase,
+    private route: ActivatedRoute,
+    private entities: EntityRegistry,
+    private entityMapper: EntityMapperService,
+    private entityFormService: EntityFormService,
+    private configService: ConfigService,
+    private entitySchemaService: EntitySchemaService,
+    private snackbar: MatSnackBar
+  ) {
+    // TODO the component should probably not handle this and it is very similar to the RemoteSession
+    this.database.initIndexedDB(
```

Comment:
what about that plan to have a RemoteSession instead of directly working with the db here? :blush: